scripts/behavior.js

```diff
@@ -371,7 +371,129 @@ function executeCommand(command, arguments, cooldown, message, settings, utils,
+                    }, rejectReason => {
+                        let specificsCreateRoleFailed = utils.getSpecifics(message.guild);
+                        delete specificsCreateRoleFailed.registeredRoles[specificsCreateRoleFailed.registeredRoles.indexOf(roleName)];
+                        utils.writeSpecifics(message.guild, specificsCreateRoleFailed);
+                        replyToMessage(message, "Could not create role due to internal discord error. " + rejectReason).then(callback(COMMAND_FORBIDDEN));
+                    });
+
+                break;
+
+            case "deleterole":
+                if (arguments == null){
+                    callback(COMMAND_MISUSE);
+                    break;
+                }
+                let delRoleName = escapeArguments(arguments);
+
+                let specificsDeleteRole = utils.getSpecifics(message.guild);
+
+                if(! specificsDeleteRole["registeredRoles"].some(role => role === delRoleName)) {
+                    replyToMessage(message, "Role not registered.")
+                        .then(callback(COMMAND_MISUSE));
+                    break;
+                }
+
+                delete specificsDeleteRole.registeredRoles[specificsDeleteRole.registeredRoles.indexOf(delRoleName)];
+                utils.writeSpecifics(message.guild, specificsDeleteRole);
+
+                let roleToDelete = Array.from(message.guild.roles.values()).find(role => role.name === delRoleName);
+                if(roleToDelete !== undefined) {
+                    roleToDelete.delete()
+                        .then(replyToMessage(message, "Role deleted").then(callback(COMMAND_SUCCESS)));
+                } else {
+                    replyToMessage(message, "Role unregistered, could not delete. Please remove the discord role manually.")
+                        .then(callback(COMMAND_SUCCESS));
+                }
+
+                break;
+
+            case "subscribe":
+                if (arguments == null){
+                    callback(COMMAND_MISUSE);
+                    break;
+                }
+                let subscribeRoleName = escapeArguments(arguments);
+
+                let specificsSubscribe = utils.getSpecifics(message.guild);
+
+                if(! specificsSubscribe["registeredRoles"].some(role => role === subscribeRoleName)) {
+                    replyToMessage(message, "Role not registered.")
+                        .then(callback(COMMAND_MISUSE));
+                    break;
+                }
+
+                let roleToSubscribe = Array.from(message.guild.roles.values()).find(role => role.name === subscribeRoleName);
+
+                if(roleToSubscribe !== undefined) {
+                    message.guild.fetchMember(message.author)
+                        .then(guildMember => {
+                            guildMember.addRole(roleToSubscribe);
+                            replyToMessage(message, "You have been subscribed to " + roleToSubscribe.name)

```

Comment:
Isn't it "You have subscribed to" instead ? 
If I were you I would remove that one and the other ("you have unsubscribed") entirely. The role a user has or not can be easily checked. Dostya doesn't need to answer in that case.

Reply:
agreed, this also prevents message spamming